internal/controllers/rest_controller.go

```diff
@@ -169,31 +195,122 @@ func (rc RESTController) UserInfo(w http.ResponseWriter, r *http.Request) {
 // Logout is responsible for revoking a token and deleting an authentication cookie.
 func (rc RESTController) Logout(w http.ResponseWriter, r *http.Request) {
 	// workflow to identify if there is a token present.
-	_, err := rc.getCredentials(r)
+	_, authType, err := rc.getCredentials(r)
 	// if a token is not found, return Forbidden.
 	if authErrors.IsTokenNotFound(err) {
-		rc.forbiddenResponse(w, err)
+		rc.forbidden(w, err)
+		return
+	}
+
+	if authType == "HeaderCredentialsHandler" {
+		rc.badRequest(w, errors.New("unable to logout without a cookie"))
+	}
```

Comment:
I didn't get this.

Reply:
🤔  requests like authorization header don't delete cookie...
well... maybe I was wrong...